File: Happy-Number/HN.py
def isHappy(n):
    numList = []
    num = n
    numListRemember = {0}

    while(1):
        numList = []
        for i in str(num):
            numList.append(int(i))
        num = sum(list(map(lambda x: x ** 2, numList)))
        if num in numListRemember:
            return False
        elif num == 1:
            return True
        else:
            numListRemember.add(num)

File: Happy-Number/HN.jl
function isHappy(s)::Bool
    a::Int = s isa Number ? s : (try
                                 parse(Int64, s)
                                 catch e
                                 throw(error("Wrong type $e"))
                                 end)
    innerHappy(a, Set([]))
end

function eachBit(n::Int)
    result = []
    while n != 0
        let left = n % 10
            push!(result, left)
            n = Int((n-left)/10)
        end
    end
    result
end

function innerHappy(n,se)
    n in se && return false
    next = sum(eachBit(n) .^ 2)
    next == 1 ? (return true) : (se = union(se, n))
    innerHappy(next,se)
end
